include/tvm/runtime/module.h

```diff
@@ -232,6 +232,8 @@ constexpr const char* tvm_param_prefix = "__tvm_param__";
 constexpr const char* tvm_lookup_linked_param = "_lookup_linked_param";
 /*! \brief The main AOT executor function */
 constexpr const char* tvm_run_func_prefix = "tvm__run_func";
+/*! \brief The entrypoint function to the generated network */
```

Comment:
let's at least update the comment here (and above, in the other constants) to explain the difference between `tvm_run_func_prefix`, `tvm_module_main`, and `tvm_entrypoint_name`. perhaps we should also qualify entrypoint (e.g. `tvm_unpacked_entrypoint_name`).

Reply:
In this PR we're generating a packed function to wrap the AOT executor function, would it be worth emitting `tvm_module_main` rather than creating a new `tvm_entrypoint_name` ?

Reply:
hrm couple thoughts:
- `tvm_module_main` is supposed to be the default name of a function contained within a runtime.Module. It tends to be used with autotuning--the lone function being tuned is named `tvm_module_main`. I believe it's also used when Relay isn't used to build a model-level thing.
- It's been called the "entrypoint" before, so it does seem like it would match
- At minimum we'd need to ensure it becomes eventually prefixed to the module. I think that's being done in https://github.com/apache/tvm/pull/8014.
- However, there is a thought from [Model-based Module Runtime Interface](https://discuss.tvm.apache.org/t/discuss-module-based-model-runtime-interface/5025) that the entrypoint should be the name of the model (as passed to tvm.relay.build). I don't know that anyone's made good on that yet, but that would be the most future-proof thing, I think (or at least, it would be most aligned with the forum).

Reply:
Regards the Model-based Module Runtime Interface, this is also being approached as part of name mangling in the RFC?

From the perspective of this PR, it makes more sense to use `tvm_module_main` and then allow the name to be mangled for use in the model-based runtime interface as part of #8014? Dependent on ordering of the merges ofc :smile_cat:

Reply:
yeah there is some overlap between Module-based Model Runtime and the firmware-facing interface. the design priority for the firmware-facing interface is that it makes sense as an embedded-facing thing. however, i'm trying to keep it to be as similar to Module-based Model Runtime interface within those constraints.

given this generated function is currently somewhat internal, i think we can use `tvm_module_main` for now and focus our efforts on the wrapping API code. should there be a case where multiple models are placed in the same module, we'll likely have to revisit that later on. but it's not clear we will support this, nor is it clear we would support it by allowing a single module to mix models. (there are, however, some compelling use cases, such as running two unrelated models in parallel).

Reply:
Sorry, what I meant was that the code to name mangle and allow multiple models is landing in #8014; that means we should use `tvm_module_main` here for now and it'll be replaced shortly by one with the model name included - thus providing the outcome we're looking for?

Also, I believe we'll see multiple models in a single embedded application with increasing frequency and we can't leave out this use case.

Reply:
yeah that seems fine to me. what i was saying was that in the multiple-model-in-one-Module case, the convention from Module-based Model Library Format would be to use the model name as the function name here. However, I think multiple models could be supported by placing them in separate Modules. I don't know of any use cases that place both models in the same Module now.